src/django/api/views/v1/production_locations.py

```diff
@@ -184,3 +188,58 @@ def create(self, request):
             },
             status=result.status_code
         )
+
+    @transaction.atomic
+    def partial_update(self, request, pk=None):
+        if switch_is_active('disable_list_uploading'):
+            raise ServiceUnavailableException(
+                APIV1CommonErrorMessages.MAINTENANCE_MODE
+            )
+        if not Facility.objects.filter(id=pk).exists():
```

Comment:
**Suggestion:** Move it ti ProductionLocationService as was implemented in https://github.com/opensupplyhub/open-supply-hub/pull/438 (ModerationEventService)